src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -37,59 +309,481 @@ const ActorInit En_Zo_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B613F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B61470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B6154C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B615E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/EnZo_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/EnZo_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B618F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B619D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B61A14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B61AF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B61CA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/EnZo_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B61FCC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/func_80B621E4.s")
+static struct_80034EC0_Entry sAnimations[] = {
+    { &D_06002FE8, 1.0f, 0.0f, -1.0f, 0, -8.0f, },
+    { &D_06002FE8, 1.0f, 0.0f, -1.0f, 0,  0.0f, },
+    { &D_06002F10, 0.0f, 1.0f,  1.0f, 2,  0.0f, },
+    { &D_06002F10, 1.0f, 1.0f, -1.0f, 0, -8.0f, },
+    { &D_06002F10, 1.0f, 8.0f, -1.0f, 0, -8.0f, },
+    { &D_0600219C, 1.0f, 0.0f, -1.0f, 0, -8.0f, },
+    { &D_06000598, 1.0f, 0.0f, -1.0f, 0, -8.0f, },
+    { &D_06000D48, 1.0f, 0.0f, -1.0f, 0, -8.0f, },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zo/EnZo_Draw.s")
+// Splashes
+void EnZo_SpawnSplashes(EnZo* this) {
+    Vec3f pos;
+    Vec3f vel;
+    s32 i;
+
+    // Convert 20 particles into splashes (all of them since there are only 15)
+    for (i = 0; i < 20; i++) {
+        f32 speed = (Math_Rand_ZeroOne() * 1.5f) + 0.5f;
+        f32 angle = Math_Rand_ZeroOne() * 6.28f; // ~pi * 2
+
+        vel.y = (Math_Rand_ZeroOne() * 3.0f) + 3.0f;
+
+        vel.x = sinf(angle) * speed;
+        vel.z = cosf(angle) * speed;
+
+        pos = this->actor.posRot.pos;
+        pos.x += vel.x * 6.0f;
+        pos.z += vel.z * 6.0f;
+        pos.y += this->actor.waterY;
+        EnZo_Splash(this, &pos, &vel, 0.08f);
+    }
+}
+
+u16 func_80B61024(GlobalContext* globalCtx, Actor* thisx) {
+    u16 textId;
+
+    textId = Text_GetFaceReaction(globalCtx, 0x1D);
+    if (textId != 0) {
+        return textId;
+    }
+
+    switch (thisx->params & 0x3F) {
+        case 8:
+            if (gSaveContext.eventChkInf[3] & 1) {
+                return 0x402A;
+            }
+            break;
+
+        case 6:
+            return 0x4020;
+
+        case 7:
+            return 0x4021;
+
+        case 0:
+            if (gBitFlags[20] & gSaveContext.inventory.questItems) {
```

Comment:
We have a macro+enum for this kind of checks. This one would go like this:
```suggestion
            if (CHECK_QUEST_ITEM(QUEST_ZORA_SAPPHIRE)) {
```